app/scripts/controllers/metametrics.js

```diff
@@ -258,54 +268,65 @@ export default class MetaMetricsController {
    *  view
    */
   trackPage({ name, params, environmentType, page, referrer }, options) {
-    if (this.state.participateInMetaMetrics === false) {
-      return;
+    try {
+      if (this.state.participateInMetaMetrics === false) {
+        return;
+      }
+
+      if (
+        this.state.participateInMetaMetrics === null &&
+        !options?.isOptInPath
+      ) {
+        return;
+      }
+      const { metaMetricsId } = this.state;
+      const idTrait = metaMetricsId ? 'userId' : 'anonymousId';
+      const idValue = metaMetricsId ?? METAMETRICS_ANONYMOUS_ID;
+      this.segment.page({
+        [idTrait]: idValue,
+        name,
+        properties: {
+          params,
+          locale: this.locale,
+          network: this.network,
+          chain_id: this.chainId,
+          environment_type: environmentType,
+        },
+        context: this._buildContext(referrer, page),
+      });
+    } catch (err) {
+      this._captureException(err);
     }
+  }
 
-    if (this.state.participateInMetaMetrics === null && !options?.isOptInPath) {
-      return;
+  /**
+   * submits a metametrics event, not waiting for it to complete or allowing its error to bubble up
+   * @param {MetaMetricsEventPayload} payload - details of the event
+   * @param {MetaMetricsEventOptions} [options] - options for handling/routing the event
+   */
+  trackEvent(payload, options) {
+    // validation is not caught and handled
+    this.validatePayload(payload);
+    try {
+      this.submitEvent(payload, options).catch((err) =>
+        this._captureException(err),
+      );
+    } catch (err) {
+      console.error(err);
     }
```

Comment:
Nit: I don't think this `try...catch` block is needed. Since `submitEvent` is `async`, any thrown error will be a Promise rejection. So this catch block won't be hit

```suggestion
    this.submitEvent(payload, options).catch((err) =>
      this._captureException(err),
    );
```